ibm_cloud_sdk_core/utils.py

```diff
@@ -100,16 +111,16 @@ def read_from_credential_file(service_name, separator='='):
     # File path specified by an env variable
     credential_file_path = getenv('IBM_CREDENTIALS_FILE')
 
-    # Home directory
+    # Top-level of the project directory
     if credential_file_path is None:
-        file_path = join(expanduser('~'), DEFAULT_CREDENTIALS_FILE_NAME)
+        file_path = join(
+            dirname(dirname(abspath(__file__))), DEFAULT_CREDENTIALS_FILE_NAME)
         if isfile(file_path):
             credential_file_path = file_path
 
-    # Top-level of the project directory
+    # Home directory
```

Comment:
In the other cores, we look in the "unix" home directory, as well as two different "home" directories associated with the Windows OS.
You can see that here in the Java SDK:
https://github.com/IBM/java-sdk-core/blob/e3d5330b0d9e2ef9cf084cb7c6aa97d6ae4f1fb4/src/main/java/com/ibm/cloud/sdk/core/util/CredentialUtils.java#L208

Does `expanduser('~')` handle all three of those?   If not, then we probably need to add the missing steps.

Reply:
yes https://docs.python.org/3/library/os.path.html#os.path.expanduser